folsom/src/main/java/com/spotify/folsom/client/OpCode.java

```diff
@@ -16,75 +16,93 @@
-  public static final int INCREMENTQ = 0x15;
-  public static final int DECREMENTQ = 0x16;
-  public static final int QUITQ = 0x17;
-  public static final int FLUSHQ = 0x18;
-
-  public static final int APPENDQ = 0x19;
-  public static final int PREPANDQ = 0x1a;
-  public static final int VERBOSITY = 0x1b;
-  public static final byte TOUCH = 0x1c;
-  public static final int GAT = 0x1d;
-  public static final int GATQ = 0x1e;
-  public static final int SASL_LIST_MECHS = 0x20;
-  public static final byte SASL_AUTH = 0x21;
-  public static final int SASL_STEP = 0x22;
-
-  public static final int RGET = 0x30;
-  public static final int RSET = 0x31;
-  public static final int RSETQ = 0x32;
-  public static final int RAPPEND = 0x33;
-  public static final int RAPPENDQ = 0x34;
-  public static final int RPREPEND = 0x35;
-  public static final int RPREPENDQ = 0x36;
-  public static final int RDELETE = 0x37;
-  public static final int RDELETEQ = 0x38;
-  public static final int RINCR = 0x39;
-  public static final int RINCRQ = 0x3a;
-  public static final int RDECR = 0x3b;
-  public static final int RDECRQ = 0x3c;
-
-  public static final int SET_VBUCKET = 0x3d;
-  public static final int GET_VBUCKET = 0x3e;
-  public static final int DEL_VBUCKET = 0x3f;
-
-  public static final int TAP_CONNECT = 0x40;
-  public static final int TAP_MUTATION = 0x41;
-  public static final int TAP_DELETE = 0x42;
-  public static final int TAP_FLUSH = 0x43;
-  public static final int TAP_OPAQUE = 0x44;
-  public static final int TAP_VBUCKET_SET = 0x45;
-  public static final int TAP_CHECKPOINT_START = 0x46;
-  public static final int TAP_CHECKPOINT_END = 0x47;
-
-  public static byte getKind(final byte opcode) {
+import static java.util.function.Function.identity;
+
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public enum OpCode {
+  GET(0x00),
+  SET(0x01),
+  ADD(0x02),
+  REPLACE(0x03),
+  DELETE(0x04),
+  INCREMENT(0x05),
+  DECREMENT(0x06),
+  QUIT(7),
+  FLUSH(0x08),
+  GETQ(0x09),
+  NOOP(0x0a),
+
+  VERSION(0x0b),
+  GETK(0x0c),
+  GETKQ(0x0d),
+  APPEND(0x0e),
+  PREPEND(0x0f),
+  STAT(0x10),
+  SETQ(0x11),
+  ADDQ(0x12),
+  REPLACEQ(0x13),
+  DELETEQ(0x14),
+  INCREMENTQ(0x15),
+  DECREMENTQ(0x16),
+  QUITQ(0x17),
+  FLUSHQ(0x18),
+
+  APPENDQ(0x19),
+  PREPANDQ(0x1a),
+  VERBOSITY(0x1b),
+  TOUCH(0x1c),
+  GAT(0x1d),
+  GATQ(0x1e),
+  SASL_LIST_MECHS(0x20),
+  SASL_AUTH(0x21),
+  SASL_STEP(0x22),
+
+  RGET(0x30),
+  RSET(0x31),
+  RSETQ(0x32),
+  RAPPEND(0x33),
+  RAPPENDQ(0x34),
+  RPREPEND(0x35),
+  RPREPENDQ(0x36),
+  RDELETE(0x37),
+  RDELETEQ(0x38),
+  RINCR(0x39),
+  RINCRQ(0x3a),
+  RDECR(0x3b),
+  RDECRQ(0x3c),
+
+  SET_VBUCKET(0x3d),
+  GET_VBUCKET(0x3e),
+  DEL_VBUCKET(0x3f),
+
+  TAP_MUTATION(0x41),
+  TAP_DELETE(0x42),
+  TAP_FLUSH(0x43),
+  TAP_OPAQUE(0x44),
+  TAP_VBUCKET_SET(0x45),
+  TAP_CHECKPOINT_START(0x46),
+  TAP_CHECKPOINT_END(0x47);
+
+  private static final Map<Byte, OpCode> BY_VALUE;
```

Comment:
Since this is used in the critical path, can we instead use a primitive array?
`OpCode[] BY_VALUE = new OpCode[0x50]` or something like that